x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IlmHealthIndicatorService.java

```diff
@@ -305,19 +341,43 @@ static class StagnatingIndicesFinder {
         public List<IndexMetadata> find() {
             var metadata = clusterService.state().metadata();
             var now = nowSupplier.getAsLong();
+
             return metadata.indices()
                 .values()
                 .stream()
                 .filter(metadata::isIndexManagedByILM)
                 .filter(md -> isStagnated(rules, now, md))
                 .toList();
         }
+
+        /**
+         *  We don't actually care what setting is updated, we just grab all the values from the `ClusterSettings` and recreate the rules
+         *  from scratch */
+        <T> void recreateRules(T ignored) {
+            var clusterSettings = clusterService.getClusterSettings();
+            var defaultMaxTimeOnAction = clusterSettings.get(MAX_TIME_ON_ACTION_SETTING);
+            var defaultMaxTimeOnStep = clusterSettings.get(MAX_TIME_ON_STEP_SETTING);
+            var defaultMaxRetriesPerStep = clusterSettings.get(MAX_RETRIES_PER_STEP_SETTING);
```

Comment:
Could we rename this away from `default*` as they're just the configured values ? (likely changed from default ?)

Reply:
done